docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,460 @@
+
+#### Set a Dataset from a File
+
+To create a `dataset` from a file, a training set and a label set must be provided. A validation set and a test set can be provided if needed.
+
+A label set is a file that has actual labels each in a line. Please refer to the example below.
+
+```
+airplane
+automobile
+bird
+cat
+deer
+dog
+frog
+horse
+ship
+truck
+```
+
+Each dataset except for the label set should contain raw float array data and one-hot-encoded labels.
+
+```
+[float array for data1][label(on-hot encoded) for data1][float array][label]...
+```
+
+After preparing the label set and dataset, create `dataset` as follows:
+
+```c
+int status = ML_ERROR_NONE;
+ml_train_dataset_h dataset;
+
+char *res_path = app_get_resource_path();
+char train_path[1024];
+char label_prop[1024];
+
+snprintf(train_path, sizeof(train_path), "%s/training.dat", res_path);
+snprintf(label_prop, sizeof(label_prop), "label_data=%s/label.dat", res_path);
+free(res_path);
+
+status = ml_train_dataset_create_with_file(&dataset, train_path);
+if(status != ML_ERROR_NONE) {
+  // handle error
+}
+
+status = ml_train_dataset_property(dataset, label_prop,
+                                   "buffer_size=9", NULL);
+status = ml_train_model_set_dataset(model, dataset);
+```
+
+Note that `buffer_size` can be set for dataset.
+
+#### Set a Dataset from a Generator
+
+Creating a dataset from a generator function is also possible.
+
+First, prepare a callback function.
+
+```c
+/**
+ * @brief      get data which size is batch for train
+ * @param[out] data
+ * @param[out] label
+ * @param[out] last if the data is finished
+ * @param[in] user_data private data for the callback
+ * @retval status for handling error
+ */
+int get_train_data(float **data, float **label, bool *last, void *user_data) {
+  /* code that fills data, label and last */
+  return ML_ERROR_NONE;
+}
+```
+
+And a dataset can be created from the callback function.
+
+```c
+int status = ML_ERROR_NONE;
+ml_train_dataset_h dataset;
+
+// validation and test callback can be omitted.
+status = ml_train_dataset_create_with_generator(&dataset, get_train_data,
+                                                NULL, NULL);
+```
+
+## Compile the Model
+
+Compiling a model finalizes the model with loss.
+Once compiled, any modification to the properties of the model.
+Adding layers or changing the optimizer/dataset of the model will not be permitted as well.
+
+```c
+int status = ML_ERROR_NONE;
+
+status = ml_train_model_compile(model, "loss=cross", "batch_size=9", NULL);)
+```
+
+## Train the Model
+
+Now, the model is ready to train. Train model as below.
+
+```c
+int status = ML_ERROR_NONE;
+
+status = ml_train_model_run(model, "epochs=20", "save_path=model.bin", NULL);)
+```
+
+## Destroy the Model
+
+After training, the model should be destroyed with `ml_train_model_destroy`.
+Note that `ml_train_model_add_layer`, `ml_train_set_optimizer`, `ml_train_set_dataset` transferrers ownership to `model`.
```

Comment:
Please change the sentence structure and use Note tag

Reply:
Done